# Anova_test

ANOVA is a parametric statistical technique that helps in finding out if there is a significant difference between the mean of three or more groups. It checks the impact of various factors by comparing groups (samples) on the basis of their respective mean. 

We can use this only when: 

the samples have a normal distribution.
the samples are selected at random and should be independent of one another.
all groups have equal standard deviations.

## One Way Anova
It is a type of hypothesis test where only one factor is considered. We use F-statistic to perform a one-way analysis of variance.One-Way ANOVA in Python: One-way ANOVA (also known as “analysis of variance”) is a test that is used to find out whether there exists a statistically significant difference between the mean values of more than one group.

Hypothesis involved:
A one-way ANOVA has the below given null and alternative hypotheses:

H0 (null hypothesis): μ1 = μ2 = μ3 = … = μk (It implies that the means of all the population are equal)
H1 (null hypothesis): It states that there will be at least one population mean that differs from the rest

Statement:

Q 1. Researchers took 20 cars of the same to take part in a study. These cars are randomly doped with one of the four-engine oils and allowed to run freely for 100 kilometers each. At the end of the journey, the performance of each of the cars is noted. Before proceeding further we need to install the SciPy library in our system. You can install this library by using the below command in the terminal:

Stepwise Implementation
Conducting a One-Way ANOVA test in Python is a step by step process and these steps are explained below:

Step 1: Creating data groups.

The very first step is to create three arrays that will keep the information of cars 

In [4]:
import numpy as np
import scipy.stats as stats
from numpy.random import randn
import seaborn as sns 
from scipy.stats import norm
import scipy

In [2]:
# Performance when each of the engine 
# oil is applied
performance1 = [89, 89, 88, 78, 79]
performance2 = [93, 92, 94, 89, 88]
performance3 = [89, 88, 89, 93, 90]
performance4 = [81, 78, 81, 92, 82]


Step 2: Conduct the one-way ANOVA:

Python provides us f_oneway() function from SciPy library using which we can conduct the One-Way ANOVA.

In [5]:
# Importing library
from scipy.stats import f_oneway

# Performance when each of the engine 
# oil is applied
performance1 = [89, 89, 88, 78, 79]
performance2 = [93, 92, 94, 89, 88]
performance3 = [89, 88, 89, 93, 90]
performance4 = [81, 78, 81, 92, 82]

# Conduct the one-way ANOVA
f_oneway(performance1, performance2, performance3, performance4)


F_onewayResult(statistic=4.625000000000002, pvalue=0.016336459839780215)

Step 3: Analyse the result:

The F statistic and p-value turn out to be equal to 4.625 and 0.016336498 respectively. Since the p-value is less than 0.05 hence we would reject the null hypothesis. This implies that we have sufficient proof to say that there exists a difference in the performance among four different engine oils.